File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map <char, int> mp;
        int l = 0;
        int res = 0;
        for (int i = 0; i < s.size(); i++) {
            if (mp.count(s[i]) == 1) {
                l = max(l, mp[s[i]]+1);
            }
            mp[s[i]] = i;
            res = max(res, i - l + 1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/meeting-schedule-ii/submission-3.py
class Solution:
    def minMeetingRooms(self, intervals: List[Interval]) -> int:
        if not intervals:
            return 0
        start = sorted([i.start for i in intervals])
        end = sorted([i.end for i in intervals])

        s_ptr = 0
        e_ptr = 0
        res = 0
        count = 0
        while s_ptr < len(start):
            if start[s_ptr] < end[e_ptr]:
                s_ptr+=1
                count+=1
            else:
                e_ptr+=1
                count-=1
            res = max(count, res)
        return res